pulseapi/utility/validators.py

```diff
@@ -0,0 +1,36 @@
+from datetime import date
+
+from django.core.exceptions import ValidationError
+from django.utils.deconstruct import deconstructible
+from django.utils.translation import gettext_lazy as _
+
+
+@deconstructible
+class YearValidator:
+    message = _('The year must be between %(min_year)s and %(max_year)s.')
+    code = 'year_value'
+
+    def __init__(self, min_offset=10, max_offset=10):
+        self.min_offset = min_offset
+        self.max_offset = max_offset
+
+    def __call__(self, value):
+        current_year = date.today().year
+        min_year = current_year - self.min_offset
```

Comment:
This is a bit of a curious system: every year this would require we "bump-down" the min_offset just to preserve the min year. In this case it feels more sensible to peg the years, rather than to bracket around the current year.

Reply:
hmm, that's a very good point! Yeah I'll peg down the current year.